### ISSUE TITLE
Running the tests of subprojects sequentially does not work

### ISSUE THREAD

**Issue by choedl** [sbt/sbt#882]
When running the tests of an aggregates subprojects, the setup methods are called immediately for all subprojects. See http://stackoverflow.com/questions/18895063/howto-run-forked-jvms-sequentially-when-testing for more information.

**Comment by choedl** [comment]
So I got my answer on stackoverflow, I had to set parallelExecution in Global := false. Should this be mentioned in the docs, or is this clear to everyone (except me)?

**Comment by harrah** [comment]
I believe the original bug could be legitimate, although I haven't looked into it yet.  Setting `parallelExecution in Global := false` disables all parallelism in all tasks.  It is just a workaround.

**Comment by thomasschoeftner** [comment]
Unfortunately, that work-around does not do the trick for me either (using sbt 0.13.0).

After modifying Chris' gist (at https://gist.github.com/choedl/6634961) to `parallelExecution in Global := false` the Test-Setup tasks are still invoked for all aggregated sub-projects concurrently instead of sequentially.

**Comment by jsuereth** [comment]
Just a note that sequentially != in the same thread.  i'm going to try to reproduce now, with a slightly altered test-case.

**Comment by jsuereth** [comment]
Alright, so it appears the issue is more around "startup" and "shutdown".  

Currently in sbt, you are not guaranteed that "startup/shutdown" for a project will finish before the next test suite starts.   These are inside different tasks, and all tasks are executed in a graph-traversal.  SO, while the tasks are not operating at the same time, you will see one project start, and then another project attempt to start.

This looks like something we'll have to invest a bit more time hunting down a solution for.

**Comment by jsuereth** [comment]
Current test case: https://gist.github.com/jsuereth/cec11c14412e365e90bf

**Comment by eed3si9n** [comment]
Ideally:

``` scala
sequential in aggregate := true
```

**Comment by wvandrunen** [comment]
+1, i'm also looking for this solution. I want to parallel compile and testCompile all the subprojects and test:run all the subprojects sequential. In other words I want to have parallel compilation and sequential test execution.

Because it is not possible to turn of parallelization in the project without losing parallel compilation we have we have all our tests in the root project instead in our subprojects. We setup a live database before running every (specs2) test. Because of this our tests should run in sequence.

Is it possible to make use of the sbt-sequential project to run all the test:run tasks in sequence (so first you would compile, test:compile with parallelExecution enabled for ThisBuild and then you execute the test:run task for every subproject in a sequence)?

**Comment by gzoller** [comment]
+1 I often have test suites that are "expensive" lots of setup, use lots of resources, etc.  Therefore I'd really like what @wvandrunen said--parallel compile/testCompile but a setting that allowed each spec to run sequentially.  Right now I have projects where I can testOnly each sub-project successfully but I crash and burn due to resource constraints when I just run test because the specs start running together.  "parallelExecution in Test = false" does not help this.

**Comment by toddmowen** [comment]
The new solution recommended in the [docs](giving more fine-grained control than `parallelExecution`) is:

```
concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)
```

I _think_ this only applies to the test run itself (specifically `test`, `testOnly`, and `testQuick`), and not to test compilation (which is the `compile` task in the test scope), so should be what @wvandrunen and @gzoller are asking for.

**Comment by richhx** [comment]
Even with `concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)`, I've noticed that our tests still run in parallel. The only way I've made it not run in parallel is to set `parallelExecution in Global := false`. And although it's only a little bit slower and avoids any race conditions, like said before, this just seems to mask the issue. Is anybody else seeing this issue? Maybe this is related? [2425](https://github.com/sbt/sbt/issues/2425)

**Comment by gzoller** [comment]
Yes, I see it now.  I wasn't aware that Console redirection can be thread safe, and it is, so that's no problem.  I'll rework this PR to be a simple documentation tweak with an example.

Carrying the philosophy forward, maybe it's worth a bit more abstracting where any output string is rendered so it can be overridden?  I'm thinking about the possibility of using console color markup templates, like jansi or something to highlight warning or error output.

**Comment by eed3si9n** [comment]
I am closing this due to inactivity, but if this is still relevant please open a new issue.
